### PR TITLE
V5.0/doc refactoring

### PR DESCRIPTION
**Pull request by lopezt-unity**
### Purpose of this PR

This PR corrects some missing features identified when preparing the doc with @amylouisee .
More relevant changes :
- Cut Tool : 
    stays open when selecting a new mesh in the hierarchy
    enable mesh selection in scene view if the cut is not started 
    fixing problem with undo (the cut was invalidated by undo-ing last point)
    adding buttons to start/end cut and tool 
- Shape Tool: 
    Adding smooth parameter to Arch, Cone, Cylinder, Pipe and Sphere 
    Adding tooltips to all the shapes' parameters in overlays
    Updating DrawShapeTool to use Unity gridSnapping in Unity 21.1 (removing custom one in the overlay)
    Updating Shortcut to open/close Shape tool (CTRL + SHIFT + K)
    Refactoring of ShapeComponent (ProBuilderShape) and Shape (ShapePrimitive) for more consistent namings

### Links

Related Docs : 
https://docs.google.com/document/d/123UdQ9kZWxao_5yCMOHtLtjQztW-c_liCXAu2blw1mQ
https://docs.google.com/document/d/1widl16zvkX-cW2XDWAVcAzYv2UwS2B2iQaaZA_3YYZ0

### Comments to Reviewers

[List known issues, planned work, provide any extra context for your code.]

### CODE REVIEW

File: Runtime/Shapes/ShapePrimitive.cs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 namespace UnityEngine.ProBuilder.Shapes
 {
     [System.Serializable]
-    public abstract class Shape
+    public abstract class ShapePrimitive
```
Comment (karl-) [inline Runtime/Shapes/ShapePrimitive.cs:4]: What is the reason for this change? `ShapePrimitive` is redundant - if `Shape` is too vague,  can we use `Primitive` or `Shape3D`

**Review by gytis-zasciurinskas — APPROVED** [review]
We did a peer review with Thomas. The PR looks solid. Checked the previous issues and all of them seem to be fixed.

However CutTool_CutUsing3NewVertices_TestInsertionTypes_TestCreates3Faces Test is currently failing, so we should take a look at it to see what is wrong
Unity_kUyuulPsXD

File: Runtime/Shapes/ShapePrimitive.cs (hunk 1 above)
Comment (lopezt-unity) [reply Runtime/Shapes/ShapePrimitive.cs]: I reverted that to Shape :)

File: Samples~/Polybrush.meta [hunk 2]
```diff
@@ -0,0 +1,8 @@
+fileFormatVersion: 2
+guid: 63867c13d780e894bad0da0b1b44538a
+folderAsset: yes
```
Comment (karl-) [inline Samples~/Polybrush.meta:3]: are these polybrush files meant to be included?

Comment (lopezt-unity) [reply Samples~/Polybrush.meta:3]: no indeed... removing these

**Review by gytis-zasciurinskas — APPROVED** [review]
Re-checked the tests, and all of them now pass on my machine. LGTM
Unity_rDfMJCDgc9

**Review by karl- — APPROVED** [review]
